mesh.py

```diff
@@ -13,28 +14,45 @@
         core
 '''
 class Mesh():
-    def __init__(self, bounds, delta_x, delta_y, delta_z):
+    def __init__(self, bounds, delta_x = 0, delta_y=0, delta_z=0,
+            default_material='water'):
         self._delta_axes = {'x': delta_x, 'y': delta_y, 'z': delta_z}
         self._boundary_mins = {'x': bounds.get_surface_coord('x', 'min'),
                 'y': bounds.get_surface_coord('y', 'min'),
                 'z': bounds.get_surface_coord('z', 'min')}
-        self._flux = np.zeros(( \
-                (bounds.get_surface_coord('x', 'max') \
-                - self._boundary_mins['x'])/delta_x,
-                (bounds.get_surface_coord('y', 'max') \
-                - self._boundary_mins['y'])/delta_y,
-                (bounds.get_surface_coord('z', 'max') \
-                - self._boundary_mins['z'])/delta_z))
+        
+        # create flux array
+        self._axis_sizes = {'x': 0, 'y': 0, 'z': 0}
+        for i, axis in enumerate(['x', 'y', 'z']):
+            self._axis_sizes[axis] = int(( \
+                    bounds.get_surface_coord(axis, 'max') - \
+                    self._boundary_mins[axis])/self._delta_axes[axis])
 
-    def get_cell(self, position):
+        self._flux = np.zeros((self._axis_sizes['x'],
+            self._axis_sizes['y'], self._axis_sizes['z']))
+
+        # create material array
+        self._cell_materials = [[[default_material \
+                for k in range(self._axis_sizes['z'])] \
+                for j in range(self._axis_sizes['y'])]  \
+                for i in range(self._axis_sizes['x'])]
+
+    def get_cell(self, position, direction=[0,0,0]):
         cell_num_vector = list()
         for num, axis in zip(xrange(3), ['x', 'y', 'z']):
-            cell_num = int((position[num]-self._boundary_mins[axis]) \
-                    / self._delta_axes[axis])
+            cell_num = int(round((position[num]-self._boundary_mins[axis]) \
+                    / self._delta_axes[axis], 7))
 
             # correct error if neutron is on upper boundary of the geometry
             if cell_num == self._flux.shape[num]:
                 cell_num -= 1
+
+            # correct error if neutron is on upper boundary of cell
+            if position[num] == (self._boundary_mins[axis] \
+                    + cell_num*self._delta_axes[axis]) \
+                    and direction[num] < 0:
+                        cell_num -=1
+
```

Comment:
I believe these two conditionals can (and should) be combined into one with the proper setup. This implementation might even be incorrect right now, but I haven't verified that. (ie.  what happens if the neutron is at the max boundary of the geometry?)